api/v1alpha1/tinkerbell_type.go

```diff
@@ -0,0 +1,125 @@
+	// Rufio contains all the information and spec about rufio.
+	Rufio RufioSpec `json:"rufio"`
+	// Tink contains all the information and spec about tink.
+	Tink TinkSpec `json:"tink"`
+}
+
+// ImageSpec specifies the details of a tinkerbell services images.
+type ImageSpec struct {
+	// ImageRepository is used to set the BootsSpec image repository.
+	ImageRepository string `json:"imageRepository,omitempty"`
+
+	// ImageTag is used to set the BootsSpec image tag.
+	ImageTag string `json:"imageTag,omitempty"`
+}
+
+// BootsSpec specifies the details of tinkerbell service boots.
+type BootsSpec struct {
+	// Image specifies the details of a tinkerbell services images
+	Image ImageSpec `json:"image"`
+
+	// DHCPAddress set the ip and port to listen on for DHCP.
+	DHCPAddress string `json:"dhcpAddress"`
+
+	// TrustedProxies comma separated allowed CIDRs subnets to be used as trusted proxies
+	TrustedProxies string `json:"trustedProxies"`
+
+	// FacilityCode represents the facility in use.
+	FacilityCode string `json:"facilityCode"`
+
+	// HTTPBind is the port to listen on for the serving iPXE binaries and files via HTTP.
+	HTTPBind int `json:"httpBind"`
+
+	// MirrorBaseURL the URL from where the "OSIE" or Hook kernel(s) and initrd(s) will be downloaded by netboot clients
+	MirrorBaseURL string `json:"mirrorBaseURL"`
+
+	// OSIEPathOverride override the URL where OSIE/Hook images are located
+	OSIEPathOverride string `json:"osiePathOverride"`
+
+	// PublicIP is the IP that netboot clients and/or DHCP relay's will use to reach Boots
+	PublicIP string `json:"publicIP"`
+
+	// PublicSyslogFQDN is the IP that syslog clients will use to send messages
+	PublicSyslogFQDN string `json:"publicSyslogFQDN"`
+
+	// SyslogBind is the port that syslog clients will use to send messages
+	SyslogBind int `json:"syslogBind"`
+
+	// TinkerbellGRPCAuthority is the IP:Port that a Tink worker will use for communicated with the Tink server
+	TinkerbellGRPCAuthority string `json:"tinkerbellGRPCAuthority"`
+
+	// TinkerbellTLS sets if the boots should run with TLS or not.
+	TinkerbellTLS bool `json:"tinkerbellTLS"`
+
+	// LogLevel sets the debug level for boots.
+	LogLevel string `json:"logLevel"`
+}
+
+// HegelSpec specifies the details of tinkerbell service hegel.
+type HegelSpec struct {
+	// Image specifies the details of a tinkerbell services images
+	Image ImageSpec `json:"image"`
+
+	// TrustedProxies comma separated allowed CIDRs subnets to be used as trusted proxies
+	TrustedProxies string `json:"trustedProxies"`
+}
+
+// RufioSpec specifies the details of tinkerbell service rufio.
+type RufioSpec struct {
+	// Image specifies the details of a tinkerbell services images
+	Image ImageSpec `json:"image"`
+
+	// TrustedProxies comma separated allowed CIDRs subnets to be used as trusted proxies
+	TrustedProxies string `json:"trustedProxies"`
+}
+
+// TinkSpec specifies the details of tinkerbell service tink server.
+type TinkSpec struct {
```

Comment:
is this tink server or controller?

Reply:
It should be for the Tink Server not controller. Did we always have a distinct image for the controller? I will add nested struct for both of them.